plugins/search/providers/sphinx_search/scripts/setSphinxLogServerPopulateActive.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * This script will set sphinx_log_server.populate_active column to either 0 (disabled) or 1 (enabled)
+ *
+ * We do this so that 'populateFromLog.php' script could run on a different machine than the machine that runs Sphinx-Search
+ * Because when a Sphinx server wants to do a resnap - it needs to stop 'populateFromLog.php'
+ * Up until today, it was done via the shell script that would call 'kaltura_populate.sh stop' locally
+ * But now, we can run 'populateFromLog.php' not locally and still control if it will process sphinx_log records or hold as long as sphinx_log_server.populate_active is 0
+ */
+
+if ($argc < 2)
+{
+	echo "Error: missing required parameter [ 0 | 1 ]" . PHP_EOL;
+	echo "To execute: $argv[0] [ 0 | 1 ] { sphinx_log_server.server }" . PHP_EOL;
+	exit(1);
+}
+
+if (!is_numeric($argv[1]) || !in_array($argv[1], array(0, 1)))
+{
+	echo "Error: parameter must be either 0 (disable) or 1 (enable)" . PHP_EOL;
+	exit(1);
+}
+
+define('ROOT_DIR', realpath(dirname(__FILE__) . '/../../../../../'));
+require_once(ROOT_DIR . '/infra/KAutoloader.php');
+require_once(ROOT_DIR . '/alpha/config/kConf.php');
+
+KAutoloader::addClassPath(KAutoloader::buildPath(KALTURA_ROOT_PATH, "vendor", "propel", "*"));
+KAutoloader::addClassPath(KAutoloader::buildPath(KALTURA_ROOT_PATH, "plugins", "*"));
+KAutoloader::setClassMapFilePath(kConf::get("cache_root_path") . '/sphinx/' . basename(__FILE__) . '.cache');
+KAutoloader::register();
+
+error_reporting(E_ALL);
+KalturaLog::setLogger(new KalturaStdoutLogger());
+
+$populateActiveValue = intval($argv[1]);
+$hostname = $argv[2] ?? ($_SERVER["HOSTNAME"] ?? gethostname());
```

Comment:
why if  $argv[2] is not set you take the running server?
also this condition like that is less readable in my opinion

Reply:
In populateFromLog we take `$_SERVER["HOSTNAME"] if set or gethostname();`
https://github.com/kaltura/server/blob/b69eb6bd3f9f570058378d62ed415aff85017e34/plugins/search/providers/sphinx_search/scripts/populateFromLog.php#L32-L31 

This works when both sphinx and sphinxPopulate are running on the same hostname
In world of k8s - it is likely that the name of the sphinx and the populate will be different
like "sphinx-1" and "sphinx-populate-1"

at the end of the day, this whole 'hostname' game is to find the record of "kaltura_sphinx_log" database for "sphinx_log_server.server"

**truthfully, made it as a 'preparation' - but it is likely we will not use it since we will adjust config to match relevant 'server' name (like in nvq2, irp2 and frp2 index-populate pods today)**

------------------------------------------------------------------
Hmmm . . is more readable? :)
`$hostname = isset($argv[2]) ? $argv[2] : (isset($_SERVER["HOSTNAME"]) ? $_SERVER["HOSTNAME"] : gethostname());`


I think once we get used to it, it's nice (like in JS)
"if is set take it" ?? "otherwise, take this value"

Reply:
@inbal-ben-david correcting myself (seems I forgot)
Once we detach sphinxPopulate from sphinx - we don't need /opt/kaltura/app/configurations on these machines
I think this is the scenario I planned for when support passing hostname manually